src/mlpack/methods/ann/layer/concat_impl.hpp

```diff
@@ -40,13 +40,47 @@ template<typename InputDataType, typename OutputDataType,
          typename... CustomLayers>
 Concat<InputDataType, OutputDataType, CustomLayers...>::Concat(
     const Concat& network) :
+    inputSize(network.inputSize),
     axis(network.axis),
     useAxis(network.useAxis),
     model(network.model),
-    run(network.run),
-    channels(network.channels)
+    run(network.run)
 {
   parameters.set_size(0, 0);
+
+  // Parameters to help calculate the number of channels.
+  size_t oldColSize = 1, newColSize = 1;
+  // Axis is specified and useAxis is true.
+  if (useAxis)
+  {
+    // Axis is specified without input dimension.
+    // Throw an error.
+    if (inputSize.n_elem > 0)
+    {
```

Comment:
I think this should fix the concat error. I was try to fix the same as well.

Reply:
No Concat error was because of the weight and bias in the linear layer and convolution layer thanks for the suggestion. I have fixed it thanks for the effort @kartikdutt18

Reply:
Ahh great, I was trying to add constructors to layers one at a time to figure out what went wrong. Thanks a lot.

Reply:
Yeah, nice approach actually problem was that I was not copying weight and bias in copy constructor because when we were calling forward visitor then weight and bias are not updating in case of a copy constructor.

Reply:
Hey Himanshu, what do you think about calling the other constructor using parameters from copy constructor to reduce redundancy?

Reply:
I didn't got you point can you elaborate more

Reply:
Since the code inside the copy constructor is same as the constructor for the class, maybe we can call that instead of repeating the same code. Sorry for not being clear enough, hopefully this make sense. Let me know what you think.

Reply:
Oh I understood it you mean to say that instead of repeating code we should use call `other` constructor I some cases like linear and concat we can do this but in some cases like recurrent_attention we have to copy a layer we will be using copyvisitor which might be a problem I think to keep it as it is to make code design of every layer similar what do you think?

Reply:
Hmm, I think I understand and that does make sense. I'll try to see if there is a workaround for layers like recurrent attention. If I find something I'll get back to you.

Reply:
Thanks for the effort @kartikdutt18

Reply:
I think you meant recurrent layer since recurrent_attention can be done by calling the other constructor. Hmm, Need a workaround for that, finally understood the issue.

Reply:
Yes I think you got it.

Reply:
@kartikdutt18 I think recurrent_attention, as well as GRU, and Dropconnect will require a copyVisitor You can see windows build failure. It was working without on my local machine. I will complete this PR by tomorrow let me know if want any work to do here

Reply:
I think you can move ahead with this, I might open a PR or issue later about how to handle this. Thanks a lot for this. I haven't gotten a chance to find a workaround yet, somehow my college has managed to make online classes more task-intensive than actually being there in class.